src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js

```diff
@@ -122,157 +118,61 @@
          */
         function insertVariantEditor(index, variant) {
 
+            if (vm.editors[index]) {
+                vm.editors[index].content.active = false;
+            }
+            variant.active = true;
+
             var variantCulture = variant.language ? variant.language.culture : "invariant";
+            var variantSegment = variant.segment;
 
             //check if the culture at the index is the same, if it's null an editor will be added
             var currentCulture = vm.editors.length === 0 || vm.editors.length <= index ? null : vm.editors[index].culture;
+            //check if the segment at the index is the same, if it's null an editor will be added
+            var currentSegment = vm.editors.length === 0 || vm.editors.length <= index ? null : vm.editors[index].segment;
+
+            
+            if (currentCulture !== variantCulture || currentSegment !== variantSegment) {
 
-            if (currentCulture !== variantCulture) {
                 //Not the current culture which means we need to modify the array.
                 //NOTE: It is not good enough to just replace the `content` object at a given index in the array
                 // since that would mean that directives are not re-initialized.
                 vm.editors.splice(index, 1, {
                     content: variant,
                     //used for "track-by" ng-repeat
-                    culture: variantCulture
+                    culture: variantCulture,
+                    segment: variantSegment
                 });
             }
             else {
                 //replace the editor for the same culture
                 vm.editors[index].content = variant;
             }
         }
-
-        function initVariant(variant, editorIndex) {
```

Comment:
So where has this code gone? There are vast amounts of changes in this PR so i can't really follow along very well but I fear a lot of stuff that we discovered was necessary is now being undone. There's a ton of test cases that needs to be covered regarding split views, invariant properties, editing values and ensure they are syncing between split views, ensuring that the same model is not being used between split views except for invariant properties, etc...

Reply:
This method basically added copies of other parts of the data to the variant object, to enable sub-parts of the editor to use that data. But the cause of this hack we had to maintain that data from multiple places. Resulting in an overly complex code, that is hard to read and not very open for change.

That's basically why I have chosen to refactor this. Let's have a talk to go through changes, cause I like to get your view on the whole thing.